x-pack/legacy/plugins/index_management/public/components/delete_templates_modal.tsx

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiConfirmModal, EuiOverlayMask, EuiCallOut, EuiCheckbox } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import React, { Fragment, useState } from 'react';
+import { toastNotifications } from 'ui/notify';
+import { deleteTemplates } from '../services/api';
+import { Template } from '../../common/types';
+
+export const DeleteTemplatesModal = ({
+  templatesToDelete,
+  callback,
+}: {
+  templatesToDelete: Array<Template['name']>;
+  callback: (data?: { hasDeletedTemplates: boolean }) => void;
+}) => {
+  const numTemplatesToDelete = templatesToDelete.length;
+
+  const hasSystemTemplate = Boolean(
+    templatesToDelete.find(templateName => templateName.startsWith('.'))
+  );
+
+  const handleDeleteTemplates = async () => {
+    deleteTemplates(templatesToDelete).then(({ data: { templatesDeleted, errors }, error }) => {
+      const hasDeletedTemplates = templatesDeleted && templatesDeleted.length;
+
+      if (hasDeletedTemplates) {
+        const successMessage = i18n.translate(
+          'xpack.idxMgmt.deleteTemplatesModal.successNotificationMessageText',
+          {
+            defaultMessage:
+              'Deleted {numSuccesses, number} {numSuccesses, plural, one {template} other {templates}}',
```

Comment:
this is OK, but i'd use this:

```suggestion
              'Deleted {numSuccesses, plural, one {# template} other {# templates}}',
```

the `#` will return the variable value.